packages/@aws-cdk/cloudformation-include/README.md

```diff
@@ -311,6 +311,15 @@ role.addToPolicy(new iam.PolicyStatement({
 }));
 ```
 
+You can also include the nested stack after the `CfnInclude` object was created,
+instead of doing it on construction:
+
+```ts
+parentTemplate.includeNestedStack('ChildTemplate', {
+  templateFile: 'path/to/my-nested-template.json',
+});
```

Comment:
I like the Lazy API, I think the ID as a positional argument make sense. Im not sure about the naming of the `getNestedStack` method, how about just `nestedStack`? 

Also, out of curiosity, can we have just one method that accept the ID and the path, if the  NestedStack object was already initialized it returns it, if not it initialize it?

Reply:
That seems weird though, if for every access you need to pass the path. And if you know that you don't need to pass the path, then you might as well use the accessor you know you need.

Reply:
> I like the Lazy API, I think the ID as a positional argument make sense. Im not sure about the naming of the `getNestedStack` method, how about just `nestedStack`?

I feel like we need to be consistent here. So, if we were to change `getNestedStack()` to `nestedStack()`, then we would have to also change all of the similar methods: `getResource()`, `getParameter()`, ..., to `resource()`, `parameter()`, ... as well. Is that what you want to do?

Reply:
> Also, out of curiosity, can we have just one method that accept the ID and the path, if the NestedStack object was already initialized it returns it, if not it initialize it?

So, Rico wanted to leave the construction parameter like it is today, but add the `includeNestedStack()` method. So, I don't think requiring the path is a good idea, because you would have to repeat it for nested stacks you included in the constructor:

```ts
const cfnTemplate = new cfn_inc.CfnInclude(this, 'Template', {
  templateFile: 'parent-template.json',
  includeNestedStacks: {
    'ChildStack': {
      templateFile: 'child-template.json',
    },
  },
});
const includedChild = cfnTemplate.getNestedStack('ChildStack', {
  templateFile: 'child-template.json', // <--- repeated!!!
});
```

Reply:
@rix0rrr I don't follow, can you add a code sample of what you mean? Specifically, what is "the accessor you know you need"

Reply:
> That seems weird though, if for every access you need to pass the path. 

Like @skinny85 already wrote, if for every `get` you need to pass the path:

```ts
fnTemplate.getNestedStack('ChildStack', {
  templateFile: 'child-template.json',  // <-- repeat
}).stack.node.tryFindChild('Banana');

fnTemplate.getNestedStack('ChildStack', {
  templateFile: 'child-template.json', // <-- repeat
}).stack.node.deleteChild('CDKMetadata');

new Construct(fnTemplate.getNestedStack('ChildStack', {
  templateFile: 'child-template.json', // <-- repeat
}).stack, 'Xyz');
```

Requiring that parameter every time is silly. The obvious solution to that is to make it optional, so now I can do this:

```ts
fnTemplate.getNestedStack('ChildStack', {
  templateFile: 'child-template.json', 
}).stack.node.tryFindChild('Banana');

fnTemplate.getNestedStack('ChildStack').stack.node.deleteChild('CDKMetadata');

new Construct(fnTemplate.getNestedStack('ChildStack').stack, 'Xyz');
```

Hurrah, right?

Except because the parameter is optional I can also do this:

```ts
fnTemplate.getNestedStack('ChildStack').stack.node.tryFindChild('Banana');
// ^^^^---- ILLEGAL! First call to getNestedStack *must* have the path parameter!
```

> And if you know that you don't need to pass the path, then you might as well use the accessor you know you need.

If you know that the parameter MUST be passed for the first call to `getNestedStack` (because it's necessary), and SHOULDN'T be passed for every other call (because it's ignored anyway)... then it should just probably be two different methods:

```ts
fnTemplate.initializeNestedStack('ChildStack', {
  templateFile: 'child-template.json',   // <-- required
});

fnTemplate.getNestedStack('ChildStack') // <-- absent
```